### DISCUSSION TITLE
Invalid datetime format: 1292 Incorrect date value: '...' for column

### DISCUSSION THREAD

**Discussion by francoism90** [spatie/laravel-data#151] [Q&A]
I want to use a different format, but it doesn't seem to work:
```
#[WithCast(DateTimeInterfaceCast::class, format: 'Y-m-d', type: CarbonImmutable::class)]
public mixed $start_date,
#[WithCast(DateTimeInterfaceCast::class, format: 'Y-m-d', type: CarbonImmutable::class)]
public mixed $end_date,
```
It keeps parsing as a full date, instead of the chosen `Y-m-d` format.

Anyone have idea what I'm doing wrong?

Thanks!

**Comment by juse-less** [comment]
Is it possible you're creating a `Data` object, and then create a new using `Data::toArray()`, or the data coming in implements `Illuminate\Contracts\Support\Arrayable`, or so?

I had this very error and took quite a bit of time to figure out the issue for me.
Basically what I did was to instantiate `Data` objects for data coming in from external API's we sync from.
After that I used `Data::validate()` to validate the actual data in a later step.
What happens when we call `Data::validate()` (or `Data::toArray()`) is that it'll transform the instantiated value (`CarbonImmutable` in my case as well) to a string.
This string will be formatted with `DATE_ATOM` by default.
So.. when I now do the actual validation, I'm expecting `Y-m-d`, but it's trying to validate it against the `DATE_ATOM` from the earlier step.

Regardless if this is your issue, or not, I'd recommend also adding a `Transformer` with the same format, since `Data::toArray()` will transform the properties.

```php
#[WithCast(DateTimeInterfaceCast::class, format: 'Y-m-d', type: CarbonImmutable::class)]
#[WithTransformer(DateTimeInterfaceTransformer::class, format: 'Y-m-d')]
public mixed $start_date,
#[WithCast(DateTimeInterfaceCast::class, format: 'Y-m-d', type: CarbonImmutable::class)]
#[WithTransformer(DateTimeInterfaceTransformer::class, format: 'Y-m-d')]
public mixed $end_date,
```

**Comment by bloodykheeng** [comment]
just put this in ur model

```
    protected $casts = [
        // Date fields
        'startDate' => 'datetime',
        'endDate'   => 'datetime',
    ];

```

```
class Notification extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'description',
        'startDate',
        'endDate',
        'gender',
        'status',
        'respondents_type',
        'created_by',
        'updated_by',
    ];

    protected $casts = [
        // Date fields
        'startDate' => 'datetime',
        'endDate'   => 'datetime',
    ];
```
